presto-main/src/test/java/com/facebook/presto/operator/TestDriver.java

```diff
@@ -142,6 +143,40 @@ public void testAbruptFinish()
     }
 
     @Test
+    public void testMemoryRevokeRequestedFuture()
```

Comment:
nit: `testMemoryRevokeRequested` (without "Future")

Did you try to have a test covering the problem this PR aims to fix? I.e. the abundance of listeners being created?

Reply:
I will try to write one.